### PR TITLE
internal/extsvc/github: Reuse V3Client.GetVersion in V4Client

### LINKED ISSUE #28276: github client: Cleaner separation of concerns between v3 and v4
The GitHub V4 client (GraphQL) [makes a REST API request](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/extsvc/github/v4.go?L327-335) to determine the version even though the GitHub V3 client (REST) exists in the codebase. This reflects a bad separation of concerns. 

Ideally, the GitHub V4 client should internally use the V3 client to make this request. This would eliminate boilerplate in the V4 client while we already have it in the V3 client.

### PR DESCRIPTION
**Pull request by indradhanush**
Fixes #28276. 
 
In this commit, we stop making a unique API request to /meta to
identify the GitHub Enterprise Server's version. It turns out that we
already have existing code to do this in the V3Client. As a result,
it's nicer to reuse the code with the trade-off of initialising the
V3Client within the V4Client when required.

This will not have any affect in the existing API quota usages of
the GraphQL client since the /meta endpoint is part of the REST API
itself and we continue to use the REST API with this change.

~Since docs.github.com is down at the moment with a `503:
Certificate Expired`,
[here's](https://web.archive.org/web/20210306150526/https://docs.github.com/en/enterprise-server@2.22/rest/reference/meta) an archive.org URL that worked for me
to verify that `/meta` is indeed under their REST API.~

Link to GitHub API docs: https://docs.github.com/en/enterprise-server@3.3/rest/reference/meta

### CODE REVIEW

File: internal/extsvc/github/v4.go [hunk 1]
```diff
@@ -324,29 +328,29 @@ func (c *V4Client) determineGitHubVersion(ctx context.Context) *semver.Version {
...
+	version = allMatchingSemver
+
 	if c.githubDotCom {
-		return allMatchingSemver
+		return
 	}
 
-	var resp struct {
-		InstalledVersion string `json:"installed_version"`
+	v, err := c.v3Client.GetVersion(ctx)
```
Comment (eseliger) [inline internal/extsvc/github/v4.go:342]: How about we instantiate the v3 Client in here only when it's needed?

**Review by eseliger — APPROVED** [review]

File: internal/extsvc/github/v4.go [hunk 2]
```diff
@@ -324,29 +328,29 @@ func (c *V4Client) determineGitHubVersion(ctx context.Context) *semver.Version {
 	return version
 }
 
-func (c *V4Client) fetchGitHubVersion(ctx context.Context) *semver.Version {
+// fetchGitHubVersion will attempt to identify the GitHub Enterprise Server's version.  If the
+// method is called by a client configured to use github.com, it will return allMatchingSemver.
+// Additinally if it fails to parse the version. or the API request fails with an error, it defaults
```
Comment (ryanslade) [inline internal/extsvc/github/v4.go:333]:
```suggestion
// Additionally if it fails to parse the version. or the API request fails with an error, it defaults
```

File: internal/extsvc/github/v4.go [hunk 3]
```diff
@@ -324,29 +328,29 @@ func (c *V4Client) determineGitHubVersion(ctx context.Context) *semver.Version {
...
+
 	if c.githubDotCom {
-		return allMatchingSemver
+		return
 	}
 
-	var resp struct {
-		InstalledVersion string `json:"installed_version"`
+	v, err := c.v3Client.GetVersion(ctx)
+	if err == nil {
```
Comment (ryanslade) [inline internal/extsvc/github/v4.go:343]: Should this be `err != nil`?

**Review by ryanslade — APPROVED** [review]
Agree with Erik's comment

File: internal/extsvc/github/v4.go (hunk 3 above)
Comment (indradhanush) [reply internal/extsvc/github/v4.go:343]: Ahhhhh. Thank you so much. That's why my test was panic'ing! You saved me some debugging time!

**Comment by indradhanush** [comment]
Will merge this tomorrow when I'm around in case something breaks.

**Review by eseliger — APPROVED** [review]

**Pull request merged**